### ISSUE TITLE
How to suppress CssSyntaxError?

### ISSUE THREAD

**Issue by cncolder** [stylelint/stylelint#2215]
> Describe the issue. Is it a bug or a feature request (new rule, new option, etc.)?

Just a question about configuration.
How to suppress CssSyntaxError?

Even I define `defaultSeverity: 'warning'` in my `.stylelint.js`. Stylelint still report error in result.

A warning is smooth. Error will break webpack compile, Atom linter and so on.

> Which rule, if any, is this issue related to?

> What CSS is needed to reproduce this issue?

> What stylelint configuration is needed to reproduce this issue?

```js
module.exports = {
  processors: ['stylelint-processor-styled-components'],
  extends: 'stylelint-config-standard',
  syntax: 'scss',
  // Set to warning for stylelint-webpack-plugin. Default error will break webpack compile even no error.
  defaultSeverity: 'warning'
}
```

> Which version of stylelint are you using?

`7.7.0`

> How are you running stylelint: CLI, PostCSS plugin, Node API?

`stylelint-webpack-plugin@0.4.2`

> Does your issue relate to non-standard syntax (e.g. SCSS, nesting, etc.)?

"Yes, it's related to SASS maps."

> What did you expect to happen?

"No errors to be flagged. Only warnings is enough."

> What actually happened (e.g. what warnings or errors you are getting)?

e.g. "The following warnings were flagged:"

```shell
src/App.js
 2:13  ✖  Missed semicolon   CssSyntaxError
```

**Comment by evilebottnawi** [comment]
@cncolder `CssSyntaxError` exception generated by `postcss`. Why you checked `App.js`? Can you show how you use `stylelint` for linting?

**Comment by m-allanson** [comment]
@evilebottnawi It looks like @cncolder is using [styled components](https://styled-components.com/) along with a [compatible processor](https://github.com/styled-components/stylelint-processor-styled-components) to extract the CSS from App.js.

**Comment by evilebottnawi** [comment]
@m-allanson hm, very interesting, need investigation, best solution is created test repo.

**Comment by cncolder** [comment]
@evilebottnawi That's not point. If `CssSyntaxError` is an exception. I think we need catch it and then return warning. Because user have write config `defaultSeverity: 'warning'`.

Now this error break my webpack compile flow. Stop my Atom editor linter tooltip.

**Comment by Arcanemagus** [comment]
@cncolder A `CssSyntaxError` isn't a warning, or a regular message from Stylelint, it's a _fatal_ error in parsing the CSS. From the sounds of it this is most likely an issue in the processor but the developers will need to investigate further to confirm.

**Comment by davidtheclark** [comment]
@Arcanemagus @cncolder I think there's a little confusion above. Here's what I understand to be the complaint: @cncolder has `defaultSeverity: 'warning'`, so expects that any report stylelint delivers will be a "warning", not an "error". Currently the `CssSyntaxError` comes through as an "error" regardless of the `defaultSeverity`. The fix would be to make `CssSyntaxError` deliver a "warning" only if `defaultSeverity: 'warning'`.

If that's right, I think it sounds like a reasonable request. @cncolder open to a PR changing this.

**Comment by davidtheclark** [comment]
Nope, I'm backtracking ... I don't think we should do this. The behavior of the linter when you get a `CssSyntaxError` is completely different from when you get a lint report with severity `warning`. It means that no further linting will work: if you have rules with severity of "error" that get triggered later on, those errors will be missed. Because of that, I *don't* think we should make the change. @cncolder if you'd like `CssSyntaxError`s to be swallowed, you can accomplish this by some other means (e.g. adding `|| true` to the end of the command in bash might work?).

**Comment by AndersDJohnson** [comment]
We'd like this as well.

**Comment by luiscla27** [comment]
What's the suggested workaround for this @davidtheclark?? 

In my case I'm doing this:
```
.user-custom-bg {
  background-image: url(../style/s-welcome-bg.style);
}
```
The thing is that ../style/s-welcome-bg.style doesn't exist at compilation time so the CssSyntaxError is thrown:

```
Module Error (from ./node_modules/postcss-loader/src/index.js):
(Emitted value instead of an instance of Error) CssSyntaxError: C:\path\to\project\some\some.component.scss:64:25: Can't resolve '../style/s-welcome-bg.style' in 'C:\path\to\project\some\'
```

**Comment by jeddy3** [comment]
@luiscla27 It's unlikely your error is related to stylelint. 

It's likely caused by whichever plugin in your [PostCSS pipeline](https://github.com/postcss/postcss-loader) is attempting to resolve the path.

**Comment by Ashan-Maduranga** [comment]
> What's the suggested workaround for this @davidtheclark??
> 
> In my case I'm doing this:
> 
> ```
> .user-custom-bg {
>   background-image: url(../style/s-welcome-bg.style);
> }
> ```
> 
> The thing is that` ../style/s-welcome-bg.style` doesn't exist at compilation time so the `CssSyntaxError` is thrown:
> 
> ```
> Module Error (from ./node_modules/postcss-loader/src/index.js):
> (Emitted value instead of an instance of Error) CssSyntaxError: C:\path\to\project\some\some.component.scss:64:25: Can't resolve '../style/s-welcome-bg.style' in 'C:\path\to\project\some\'
> ```

Did you find a solution?

**Comment by luiscla27** [comment]
> Did you find a solution?
I removed that line from my `scss` file and moved it to a `css`, I did that only to the `background-image` attribute
